Donation-platform/src/main/java/com/google/RecommendationSystem.java

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google;
+
+import com.google.Charity;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Arrays;
+
+/* Class that takes user-selected tags as input and finds the according best-matching charities.*/
+public class RecommendationSystem {
+
+    // Determine how much the charity's tags matching matters compared to how trending the charity is
+    private final double TAG_MATCHING_WEIGHT = 0.7;
+    private final double CHARITY_TRENDING_WEIGHT = 0.3;
+
+    // Increment constants to charity's tagValue for having the given user-selected tags (#1, #2, or #3)
+    private final double TAGSCORE_INCREMENT_FOR_HAVING_TAG_1 = 0.8;
+    private final double TAGSCORE_INCREMENT_FOR_HAVING_TAG_2 = 0.5;
+    private final double TAGSCORE_INCREMENT_FOR_HAVING_TAG_3 = 0.3;
+    private final double TAGSCORE_INCREMENT_FOR_HAVING_TAG_1_AND_2 = 0.2;
+    private final double TAGSCORE_INCREMENT_FOR_HAVING_TAG_1_AND_3 = 0.15;
+    private final double TAGSCORE_INCREMENT_FOR_HAVING_TAG_2_AND_3 = 0.1;
+
+    // Sorts charities according to the scores calculated by getCharityScores
+    public List<Charity> getBestMatches(List<String> selectedTags) {
```

Comment:
if its only doing sorting then name it sortCharities

Reply:
Sorting the charities is part of getting the best-matching charities. The other part (which comes before) is getting the mapping of charities to their respective relevance scores (which is done by calling getCharityScores).

Reply:
I will update the comment to make this more clear. I'd rather not change the name of the function to sortCharities because that only describes part of what the function does.

Reply:
Fixed with recent commit.